src/PIL/EpsImagePlugin.py

```diff
@@ -333,22 +341,38 @@ def _read_comment(s: str) -> bool:
                 else:
                     break
 
-                self._size = columns, rows
-                return
+                # Read the columns and rows after checking the bit depth and mode
+                # in case the bit depth and/or mode are invalid.
+                imagedata_size = columns, rows
             elif bytes_mv[:5] == b"%%EOF":
                 break
             elif trailer_reached and reading_trailer_comments:
                 # Load EPS trailer
                 s = str(bytes_mv[:bytes_read], "latin-1")
-                _read_comment(s)
+                read_comment(s)
             elif bytes_mv[:9] == b"%%Trailer":
                 trailer_reached = True
             bytes_read = 0
 
-        if not self.tile:
+        # A "BoundingBox" is always required,
+        # even if an "ImageData" descriptor size exists.
```

Comment:
Could you explain to me (or refer to a specific part of a specification) why this is the case?

Reply:
https://www.loc.gov/preservation/digital/formats/fdd/fdd000246.shtml

> The PostScript code that specifies an EPS image is subject to constraints, specified in the [PostScript Language Document Structuring Conventions Specification (DSC)](https://web.archive.org/web/20190126043023/https://www.adobe.com/content/dam/acom/en/devnet/actionscript/articles/5001.DSC_Spec.pdf).
> ...
> Most importantly, an EPS file must include two required DSC header comments:
> * %!PS-Adobe-3.0 EPSF-3.0
> * '%%BoundingBox: llx lly urx ury'

Reply:
Ok, I see in '2.1 Required DSC Header Comments' of https://web.archive.org/web/20220120164601/https://www.adobe.com/content/dam/acom/en/devnet/actionscript/articles/5002.EPSF_Spec.pdf
> The two required DSC Header comments are
> > %!PS-Adobe-3.0 EPSF-3.0
> > %%BoundingBox: llx lly urx ury

That same PDF also includes in 3.1
> If either of the two required header comments is missing, the application should alert the user that the file is not a conforming EPS file and abort the import.